thumbor/handlers/__init__.py

```diff
@@ -258,23 +269,22 @@ def _load_media(self, context):
 
         quality = self.context.request.quality
         if quality is None:
-            if image_extension == '.webp' and self.context.config.WEBP_QUALITY is not None:
+            if image_extension == '.webp' and \
+                            self.context.config.WEBP_QUALITY is not None:
                 quality = self.context.config.get('WEBP_QUALITY')
             else:
                 quality = self.context.config.QUALITY
 
-
-        media = None
-
         logger.debug('Reading image as {extension}'.format(
             extension=image_extension
         ))
 
         try:
             result = context.request.engine.read(image_extension, quality)
-            media = Media.from_result(result)
+            media = Media(result, mimetype=content_type)
         except Exception as e:
-            logger.error(e)
+            # logger.error(e)
+            logger.exception('BaseHandler:_load_media failed:')
```

Comment:
The exception is in the context. With logger.exception the stacktrace is logged as well. Although we should probably put it back to error once everything is working.